src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -323,6 +323,32 @@ internal static class ConstraintsHelper
             diagnostics.Add(location, useSiteDiagnostics);
         }
 
+        internal static ImmutableArray<TypeParameterConstraintClause> MakeTypeParameterConstraints(
+            this SourceMethodSymbol containingSymbol,
+            ImmutableArray<TypeParameterSymbol> typeParameters,
+            SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses,
+            Location location,
+            DiagnosticBag diagnostics)
+        {
+            if (typeParameters.Length == 0 || constraintClauses.Count == 0)
+            {
+                return ImmutableArray<TypeParameterConstraintClause>.Empty;
+            }
+
+            var compilation = containingSymbol.DeclaringCompilation;
+            var binderFactory = compilation.GetBinderFactory(constraintClauses[0].SyntaxTree);
+            var binder = binderFactory.GetBinder(constraintClauses[0]);
```

Comment:
>var binder = binderFactory.GetBinder(constraintClauses[0]); [](start = 12, length = 59)

I am afraid this probably won't give us correct binder when we are operating on a local function. We probably do not have any tests where a type parameter is constrained on another type parameter.